operator/builtin/transformer/filter/filter.go

```diff
@@ -96,7 +101,17 @@ func (f *FilterOperator) Process(ctx context.Context, entry *entry.Entry) error
 		return nil
 	}
 
-	if !filtered || rand.Float64() > f.dropRatio {
+	if !filtered {
+		f.Write(ctx, entry)
+		return nil
+	}
+
+	i, err := randInt(rand.Reader, upperBound)
```

Comment:
Is there a behavior change here? I am having a hard time establishing equivalence with previous version of the code. I assume there is no intent to change the behavior, just fix the gosec warnings?

Reply:
There's not intended to be. This was my first time using `crypto/rand`, I a found it to be surprisingly cumbersome to use compared to `math/rand`, though I understand it's best practice. Because I had to make some changes, I also split apart this compound conditional. I believe the net result is the same.